bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeDiscoveryService.java

```diff
@@ -0,0 +1,192 @@
+                                            continue;
+                                        }
+                                        throw e;
+                                    }
+                                    break;
+                                }
+                            }
+                        } catch (IOException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+                            logger.warn("Error connecting to device at {}: {}", currentIp, e.getMessage());
+                        }
+                    }
+                }
+            }
+        } catch (SocketException | InterruptedException e) {
+            logger.warn("Error while discovering: {}", e.getMessage());
+        }
+    }
+
+    private void discoverWithADB(String ip, int port)
+            throws InterruptedException, IOException, AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException,
+            AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException {
+        var device = new AndroidDebugBridgeDevice();
+        device.configure(ip, port);
+        try {
+            device.connect();
+            logger.debug("connected adb at {}:{}", ip, port);
+            String serialNo = device.getSerialNo();
+            String model = device.getModel();
+            String androidVersion = device.getAndroidVersion();
+            String brand = device.getBrand();
+            logger.debug("discovered: {} - {} - {} - {}", model, serialNo, androidVersion, brand);
+            onDiscoverResult(serialNo, ip, port, model, androidVersion, brand);
+        } finally {
+            device.disconnect();
+        }
+    }
+
+    @Override
+    protected void stopScan() {
+        super.stopScan();
+        discoveryRunning = false;
+        logger.debug("scan stopped");
+    }
+
+    private void onDiscoverResult(String serialNo, String ip, int port, String model, String androidVersion,
+            String brand) {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put(Thing.PROPERTY_SERIAL_NUMBER, serialNo);
+        properties.put(PARAMETER_IP, ip);
+        properties.put(PARAMETER_PORT, port);
+        properties.put(Thing.PROPERTY_MODEL_ID, model);
+        properties.put(Thing.PROPERTY_VENDOR, brand);
+        properties.put(Thing.PROPERTY_FIRMWARE_VERSION, androidVersion);
+        thingDiscovered(DiscoveryResultBuilder.create(new ThingUID(THING_TYPE_ANDROID_DEVICE, serialNo))
+                .withTTL(DISCOVERY_RESULT_TTL_SEC).withRepresentationProperty(Thing.PROPERTY_SERIAL_NUMBER)
+                .withProperties(properties).withLabel(String.format("%s (%s)", model, serialNo)).build());
+    }
+
+    @Reference
+    void setConfigurationAdmin(ConfigurationAdmin admin) {
+        this.admin = admin;
+    }
```

Comment:
Great to see this works!

You could inject this via the constructor to get rid of the Nullable annotation.